jwt/algorithms.py

```diff
@@ -163,6 +199,31 @@ def compute_hash_digest(self, bytestr: bytes) -> bytes:
         else:
             return bytes(hash_alg(bytestr).digest())
 
+    def check_crypto_key_type(self, key: PublicKeyTypes | PrivateKeyTypes):
+        """Check that the key belongs to the right cryptographic family.
+
+        Note that this method only works when `cryptography` is installed.
+
+        Args:
+            key (Any): Potentially a cryptography key
+        Raises:
+            ValueError: if `cryptography` is not installed, or this method is called by a non-cryptography algorithm
+            InvalidKeyError: if the key doesn't match the expected key classes
+        """
+        if not has_crypto or self._crypto_key_types is None:
+            raise ValueError(
+                "This method requires the cryptography library, and should only be used by cryptography-based algorithms."
+            )
+
+        # TODO check for algo_type? (e.g., SHA256 vs SHA384)
```

Comment:
this is more of a question (should we check algo type?), since the current checks don't do that

Reply:
will that add more value?

Reply:
I guess we should try

Reply:
hmm, @auvipy I gave this a try, but I couldn't find an easy way to determine hash type to compare to in this case. I also don't remember what my initial idea was for this (maybe I thought it'd be easier). I vote we skip this for now.